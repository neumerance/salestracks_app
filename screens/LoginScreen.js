import React from 'react';
import { View, Text, StyleSheet } from 'react-native';
import { Input, Button } from 'react-native-elements';
import Icon from 'react-native-vector-icons/FontAwesome';
import { COMPANY_NAME, COMPANY_LINE } from 'react-native-dotenv'


class LoginScreen extends React.Component {
  static navigationOptions = {
    header: null,
  };

  constructor(props) {
    super(props);
    this.state = {
      email: '',
      password: ''
    }
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.screenTitles}>
          <Text style={styles.title}>{COMPANY_NAME}</Text>
          <Text>{COMPANY_LINE}</Text>
        </View>
        <View>
          <Input
            style={styles.textInput}
            placeholder='Email'
            shake={true}
            leftIcon={
              <Icon
                name='user'
                size={24}
                color='black'
              />
            }
          />
        </View>
        <View>
          <Input
            style={styles.textInput}
            placeholder='Password'
            shake={true}
            secureTextEntry={true}
            leftIcon={
              <Icon
                name={'key'}
                size={24}
                color='black'
              />
            }
          />
        </View>
        <View style={styles.loginButton}>
          <Button
            onPress={this.onLogin}
            title="Login"
            color="#841584"
            accessibilityLabel="Authenticate user"
          />
        </View>
      </View>
    )
  }

  onLogin() {

  }

  setEmail(e) {
    this.setFieldValue(e.target.value, 'email');
  }

  setPassword(e) {
    this.setFieldValue(e.target.value, 'password');
  }

  setFieldValue(value, field) {
    const state = this.state;
    state[field] = value;
    this.setState(state);
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center'
  },
  screenTitles: {
    alignItems: 'center',
    marginBottom: 30
  },
  title: {
    fontSize: 24
  },
  loginButton: {
    marginTop: 20,
    marginLeft: 10,
    marginRight: 10
  }
})

export default LoginScreen;
