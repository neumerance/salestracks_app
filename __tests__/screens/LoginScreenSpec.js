import React from 'react';
import { shallow } from 'enzyme';
import LoginScreen from '../../screens/LoginScreen';

describe('LoginScreen', () => {
  it('renders correctly', () => {
    const component = shallow(<LoginScreen />);
    expect(component).toMatchSnapshot();
  });
});
