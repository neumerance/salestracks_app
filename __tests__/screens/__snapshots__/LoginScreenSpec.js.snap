// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 24,
        }
      }
    >
      Salestracks
    </Text>
    <Text>
      We move business faster
    </Text>
  </View>
  <View>
    <ForwardRef(Themed.Input)
      editable={true}
      leftIcon={
        <Icon
          allowFontScaling={false}
          color="black"
          name="user"
          size={24}
        />
      }
      onChangeText={[Function]}
      placeholder="Email"
      shake={true}
      value=""
    />
  </View>
  <View>
    <ForwardRef(Themed.Input)
      editable={true}
      leftIcon={
        <Icon
          allowFontScaling={false}
          color="black"
          name="key"
          size={24}
        />
      }
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      shake={true}
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginLeft": 10,
        "marginRight": 10,
        "marginTop": 20,
      }
    }
  >
    <ForwardRef(Themed.Button)
      accessibilityLabel="Authenticate user"
      color="#841584"
      loading={false}
      onPress={[Function]}
      title="Login"
    />
  </View>
</View>
`;
